membership/cb_membership.go

```diff
@@ -0,0 +1,261 @@
+package membership
+
+import (
+	"context"
+	"errors"
+	"time"
+
+	jsoniter "github.com/json-iterator/go"
+
+	"github.com/Trendyol/go-dcp-client/dcp"
+
+	"github.com/google/uuid"
+
+	"github.com/Trendyol/go-dcp-client/helpers"
+	"github.com/Trendyol/go-dcp-client/logger"
+	"github.com/Trendyol/go-dcp-client/membership/info"
+	"github.com/couchbase/gocbcore/v10"
+	"github.com/couchbase/gocbcore/v10/memd"
+)
+
+type cbMembership struct {
+	info                *info.Model
+	infoChan            chan *info.Model
+	client              dcp.Client
+	id                  []byte
+	clusterJoinTime     int64
+	lastActiveInstances []Instance
+	handler             info.Handler
+	monitorQuery        []byte
+	indexQuery          []byte
+}
+
+type Instance struct {
+	ID              *string `json:"id,omitempty"`
+	Type            string  `json:"type"`
+	HeartbeatTime   int64   `json:"heartbeatTime"`
+	ClusterJoinTime int64   `json:"clusterJoinTime"`
```

Comment:
It would be better if we add a field "hertbeatIntervalInMillis" to use while calculating a instance is alive or not. We'll treat each instance according to their heartbeat send periods. Basically you'll pass this config information while registering yourself to the registry(while creating the instance document).

Reply:
https://github.com/Trendyol/go-dcp-client/blob/2d87ae75bc3b48afd8046f863583a858b7d9d89c/membership/cb_membership.go#L43

I assume all instances need to run with the same code, so I added it as a constant.